packages/shared/src/components/modals/ShareModal.tsx

```diff
@@ -31,9 +31,12 @@ export default function ShareModal({
   ...props
 }: ShareModalProps): ReactElement {
   const isComment = !!comment;
+  const permalink = isComment
+    ? post?.commentsPermalink
+    : post?.sharedPost?.commentsPermalink ?? post?.commentsPermalink;
```

Comment:
Why do you need this one? ?? `post?.commentsPermalink`

Reply:
When `isComment` is true, it would mean the user clicked the comment's share button. I think the intent is most likely to share the comment itself not the article this time.

Reply:
Did you want to have something like this?
```
  const permalink = isComment || !post?.sharedPost
    ? post?.commentsPermalink
    : post?.sharedPost?.commentsPermalink;
```

Reply:
I think the current setup is fine. That case should only be for `isComment` to true. When false, we always want to return the `sharedPost` link when available (to prioritize the actual post link).